x/coin_test.go

```diff
@@ -340,16 +341,17 @@ func TestCoinDivide(t *testing.T) {
 
 func TestCoinMultiply(t *testing.T) {
 	cases := map[string]struct {
-		coin  Coin
-		times int64
-		want  Coin
+		coin    Coin
+		times   int64
+		want    Coin
+		wantErr error
 	}{
 		"zero value coin": {
 			coin:  NewCoin(0, 0, "DOGE"),
 			times: 666,
 			want:  NewCoin(0, 0, "DOGE"),
 		},
-		"multiply": {
+		"simple multiply": {
```

Comment:
Add a case for fractional flowing into whole:

```golang
{
			coin:  NewCoin(1, 230000000, "DOGE"),
			times: 10,
			want:  NewCoin(12, 300000000, "DOGE"),
		},
```

Reply:
There is one already, but I will add this one as well.